File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 9/14/2025, 9:19:11 AM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs)==0:
            return ""
        prefix=strs[0]
        for i in range(len(strs)):
            while(strs[i].find(prefix)!=0):
                prefix=prefix[0:len(prefix)-1]
                if len(prefix)==0:
                    return ""
        return prefix   

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 9/14/2025, 9:19:06 AM
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);

        String first = strs[0];
        String last = strs[strs.length-1];

        int len = first.length();

        int index = 0;
        for(int i=0;i<len;i++){
            char c1 = first.charAt(i);
            char c2 = last.charAt(i);
            if(c1 == c2){
                index++;
            }else{
                break;
            }
        }

        return index==0?"":first.substring(0,index);
    }
}